### PR TITLE
To add single and chained learning schedulers to docs

### PR DESCRIPTION
**Pull request by iramazanli**
In the optimizer documentation, many of the learning rate schedulers [examples](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate) are provided according to a generic template. In this PR we provide a precise simple use case example to show how to use learning rate schedulers. Moreover, in a followup example we show an example how to chain two schedulers next to each other.

### CODE REVIEW

File: docs/source/optim.rst [hunk 1]
```diff
@@ -146,12 +146,45 @@ allows dynamic learning rate reducing based on some validation measurements.
...
+In many places during this tutorial, we will use the following template to refer to schedulers
+algorithms.
+
     >>> scheduler = ...
     >>> for epoch in range(100):
     >>>     train(...)
     >>>     validate(...)
     >>>     scheduler.step()
 
+
```
Comment (vincentqb) [inline docs/source/optim.rst]: nit: does this change the formatting?

File: docs/source/optim.rst [hunk 2]
```diff
@@ -146,6 +146,38 @@ allows dynamic learning rate reducing based on some validation measurements.
...
+        for input, target in dataset:
+            optimizer.zero_grad()
+            output = model(input)
+            loss = loss_fn(output, target)
+            loss.backward()
+            optimizer.step()
+            scheduler.step()
+
+One interesting feature of schedulers are that they can be called back-to-back and to
+get chained. This will lead to by schedulers are getting applied simultaneously.
```
Comment (vincentqb) [inline docs/source/optim.rst:162]: nit: I'd suggest removing "interesting feature" and highlighting that "most schedulers" support that feature (but not all). "simultaneously" also seems unclear to me: does it mean one override the other? I suggest:

> Most learning rate schedulers can be called back-to-back (also referred to as chaining schedulers). The result is that each scheduler is applied one after the other on the learning rate obtained by the one preceding it.

File: docs/source/optim.rst [hunk 3]
```diff
@@ -146,6 +146,38 @@ allows dynamic learning rate reducing based on some validation measurements.
...
+
+    scheduler = ExponentialLR(optimizer, gamma=0.9)
+    for epoch in range(20):
+        for input, target in dataset:
+            optimizer.zero_grad()
+            output = model(input)
+            loss = loss_fn(output, target)
+            loss.backward()
+            optimizer.step()
+            scheduler.step()
```
Comment (vincentqb) [inline docs/source/optim.rst:159]: thanks for adding this code example! i like that it emphasize that scheduler step is done after the optimizer step.

File: docs/source/optim.rst [hunk 4]
```diff
@@ -146,6 +146,38 @@ allows dynamic learning rate reducing based on some validation measurements.
 Learning rate scheduling should be applied after optimizer's update; e.g., you
 should write your code this way:
 
+Example::
```
Comment (vincentqb) [inline docs/source/optim.rst:149]: good idea to adjust the formatting and be more consistent :)

File: docs/source/optim.rst (hunk 3 above)
Comment (vincentqb) [reply docs/source/optim.rst:159]: nit: though some schedulers are applied only at every epoch. do you have a way of suggesting this without adding too much here? if it's cumbersome, we don't need to make that point :)

File: docs/source/optim.rst [hunk 5]
```diff
@@ -146,6 +146,38 @@ allows dynamic learning rate reducing based on some validation measurements.
...
+        for input, target in dataset:
+            optimizer.zero_grad()
+            output = model(input)
+            loss = loss_fn(output, target)
+            loss.backward()
+            optimizer.step()
+            scheduler1.step()
+            scheduler2.step()
+
+In many places during this tutorial, we will use the following template to refer to schedulers
```
Comment (vincentqb) [inline docs/source/optim.rst:178]: nit: this will be part of the documentation, and not a tutorial :)
```suggestion
In many places in the documentation, we will use the following template to refer to schedulers
```

File: docs/source/optim.rst (hunk 3 above)
Comment (iramazanli) [reply docs/source/optim.rst]: Thanks for the amazing review :) and i did a little simplification. Hope you're liking the current version.

File: docs/source/optim.rst (hunk 1 above)
Comment (iramazanli) [reply docs/source/optim.rst]: actually this empty line was unnecessary, so deleted it :)

File: docs/source/optim.rst [hunk 6]
```diff
@@ -146,6 +146,35 @@ allows dynamic learning rate reducing based on some validation measurements.
...
 
+Example::
+
+    model = [Parameter(torch.randn(2, 2, requires_grad=True))]
+    optimizer = SGD(model, 0.1)
+    scheduler = ExponentialLR(optimizer, gamma=0.9)
+
+    for epoch in range(20):
+        optimizer.step()
+        scheduler.step()
```
Comment (vincentqb) [inline docs/source/optim.rst:157]: Observations:
- It's easier to compare different sections of the documentation if they are as similar as possible.
- It's important to tell users where the rest of the training code would go relative to the scheduler step.
- The optimizer is stepped at every dataset iteration, not just epoch.

This is a little verbose but bridges the two types of examples presented on the documentation page.
```
    model = [Parameter(torch.randn(2, 2, requires_grad=True))]
    optimizer = SGD(model, 0.1)
    scheduler = ExponentialLR(optimizer, gamma=0.9)

    for epoch in range(20):
        for input, target in dataset:
            optimizer.zero_grad()
            output = model(input)
            loss = loss_fn(output, target)
            loss.backward()
            optimizer.step()
        scheduler.step()
```

Comment (iramazanli) [reply docs/source/optim.rst]: Yes, indeed this looks much more consistent with the rest of the documentation :)

To preserve consistency even more i modified chain example as well accordingly. Hope it was a right decision :)

**Review by vincentqb — APPROVED** [review]
LGTM, thanks for updating the doc!
